pysal/contrib/viz/color.py

```diff
@@ -0,0 +1,178 @@
+                      colormap scale type  [sequential, diverging, qualitative]
+
+    k               : int
+                      number of classes
+
+    color_encoding  : string
+                      encoding of colors [hexc, rgb, mpl, mpl_colormap]
+                      * hex: list of hex strings
+                      * rgb: list of RGB 0-255 triplets
+                      * mpl: list of RGB 0-1 triplets as used by matplotlib
+                      * mpl_colormap: matplotlib color map
+
+    Returns
+    -------
+
+    colors:  color map in the specified color_encoding
+
+    """
+    encs = {'hex': 'hex_colors',
+            'rgb': 'colors',
+            'mpl': 'mpl_colors',
+            'mpl_colormap': 'mpl_colormap'}
+    if not palette:
+        cmtype = pltt2type[name.lower()]
+        if name[-2:] == '_r':
+            palette = pltt.colorbrewer.get_map(name[:-2], cmtype,
+                    k, reverse=True)
+        else:
+            palette = pltt.colorbrewer.get_map(name, cmtype, k)
+    colors = getattr(palette, encs[color_encoding.lower()])
+    return colors
+
+def _build_pltt2type():
+    types = ['sequential', 'diverging', 'qualitative']
+    pltt2type = {}
+    for t in types:
+        pals = list(set([
+            p.split('_')[0] for p in dir(getattr(pltt.colorbrewer, t)) 
+                            if p[0]!='_'
+                            ]))
+        pals = pals + [p+'_r' for p in pals]
+        for p in pals:
+            pltt2type[p.lower()] = t
+    return pltt2type
+
+pltt2type = _build_pltt2type()
+
+def get_color_map_b2m(name='BuGn', cmtype='sequential', k=5,
+                  color_encoding='hexc'):
+
+    """
+    DEPRECATED!!!
+
+    Get a brewer colormap from `brewer2mpl`
+
+    Arguments
+    ---------
+
+    name:   string
+            colormap name
+
+    cmtype: string
+            colormap scale type  [sequential, diverging, qualitative]
+
+    k:  int
+        number of classes
+
+    color_encoding: string
+                    encoding of colors [hexc, rgb, mpl, mpl_colormap]
+                    hex: list of hex strings
+                    rgb: list of RGB 0-255 triplets
+                    mpl: list of RGB 0-1 triplets as used by matplotlib
+                    mpl_colormap: matplotlib color map
+
+    Returns
+
+    colors:  color map in the specified color_encoding
+
+    """
+    encs = {'hexc': 'hex_colors',
+            'rgb': 'colors',
+            'mpl': 'mpl_colors',
+            'mpl_colormap': 'mpl_colormap'}
+    try:
+        bmap = brewer2mpl.get_map(name, cmtype, k)
+        colors = getattr(bmap, encs[color_encoding.lower()])
+        return colors
+    except:
+        print('Color map not found: ', name, cmtype, k)
+
+def get_maps_by_type(data_type):
+    names = [name for name in dir(data_type) if not name.startswith('_')]
+    return names
+
+# get maps for each ctype and default k=5 for populating display options
+ctypes = (b2m_sequential, b2m_diverging, b2m_qualitative)
```

Comment:
that was supposed to be me not my evil twin.

On Mon, Aug 15, 2016 at 3:52 PM, sjsreyeviltwin <notifications@github.com>
wrote:

> In pysal/contrib/viz/color.py
> <https://github.com/pysal/pysal/pull/844#discussion_r74850272>:
>
> > +            'rgb': 'colors',
> > +            'mpl': 'mpl_colors',
> > +            'mpl_colormap': 'mpl_colormap'}
> > +    try:
> > +        bmap = brewer2mpl.get_map(name, cmtype, k)
> > +        colors = getattr(bmap, encs[color_encoding.lower()])
> > +        return colors
> > +    except:
> > +        print('Color map not found: ', name, cmtype, k)
> > +
> > +def get_maps_by_type(data_type):
> > +    names = [name for name in dir(data_type) if not name.startswith('_')]
> > +    return names
> > +
> > +# get maps for each ctype and default k=5 for populating display options
> > +ctypes = (b2m_sequential, b2m_diverging, b2m_qualitative)
>
> I have a pr in that went with option B).
>
> Also I think the example notebook has some broken cells:
> https://github.com/darribas/pysal/blob/20530cf26552ed6c5684f3388ee78c
> c022d52b5a/pysal/contrib/viz/geotable_plot.ipynb
>
> See In[13]
>
> colors = colors.map({i:j for i,j in enumerate(pltt)})
>
> but before this in In [22] pltt is a module so In [12] fails for me. I
> think you may have been going back and forth as it looks like a case of
> non-linear editing (i.e., maybe 22 changed after 12 even though 12 is down
> further in the notebook? )
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pysal/pysal/pull/844/files/20530cf26552ed6c5684f3388ee78cc022d52b5a#r74850272>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHdywN5TfEVR7fd7WijybzlzWpsn0hUcks5qgO3EgaJpZM4JO6Rn>
> .
>



-- 
Sergio (Serge) Rey
Professor of Computational Spatial Science
School of Geographical Sciences and Urban Planning
Center for Geographical Information Science
Arizona State University
https://geoplan.asu.edu/sergio-rey

Editor, Geographical Analysis
http://wileyonlinelibrary.com/journal/gean